trl/trainer/ddpo_config.py

```diff
@@ -4,107 +4,138 @@
 from dataclasses import dataclass, field
 from typing import Literal, Optional
 
-from ..core import flatten_dict
+from transformers import TrainingArguments
+
 from ..import_utils import is_bitsandbytes_available, is_torchvision_available
 
 
 @dataclass
-class DDPOConfig:
-    """
-    Configuration class for DDPOTrainer
+class DDPOConfig(TrainingArguments):
+    r"""
+    Configuration class for the [`DDPOTrainer`].
+
+    Using [`~transformers.HfArgumentParser`] we can turn this class into
+    [argparse](https://docs.python.org/3/library/argparse#module-argparse) arguments that can be specified on the
+    command line.
+
+    Args:
+        exp_name (`str`, *optional*, defaults to `os.path.basename(sys.argv[0])[: -len(".py")]`):
+            Name of this experiment (by default is the file name without the extension name).
+        log_with (`Optional[Literal["wandb", "tensorboard"]]`, *optional*, defaults to `None`):
+            Log with either 'wandb' or 'tensorboard', check
+            https://huggingface.co/docs/accelerate/usage_guides/tracking for more details.
+        tracker_kwargs (`Dict`, *optional*, defaults to `{}`):
+            Keyword arguments for the tracker (e.g. wandb_project).
+        accelerator_kwargs (`Dict`, *optional*, defaults to `{}`):
+            Keyword arguments for the accelerator.
+        project_kwargs (`Dict`, *optional*, defaults to `{}`):
+            Keyword arguments for the accelerator project config (e.g. `logging_dir`).
+        tracker_project_name (`str`, *optional*, defaults to `"trl"`):
+            Name of project to use for tracking.
+        logdir (`str`, *optional*, defaults to `"logs"`):
+            Top-level logging directory for checkpoint saving.
+        num_epochs (`int`, *optional*, defaults to `100`):
```

Comment:
Technically I think one could reuse the `num_train_epochs` from the parent class?

Reply:
Yes, I agree 100%. But I try not to deal with these aspects in this PR, to keep the changes atomic, non-breaking, and easy to review. 
The only exception is that, as I want the configurations to inherit from `TrainingArguments`, I have to deal with cases where some arguments overwrite those of `TrainingArguments`. For all other cases, I leave them for a separate PR.

Reply:
Sounds good to me!